.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
you should rather specify the paths instead of branches. whenever your project files are changed